ngraph/core/src/function.cpp

```diff
@@ -520,3 +516,78 @@ ov::op::util::Variable::Ptr ov::Function::get_variable_by_id(const string& varia
+    return m_results.at(i);
+}
+ov::Output<ov::Node> ov::Function::output(const std::string& tensor_name) const {
+    for (const auto& res : m_results) {
+        for (const auto& name : res->get_input_tensor(0).get_names()) {
+            if (name == tensor_name)
+                return res;
+        }
+    }
+    throw ov::Exception("Output for tensor name " + tensor_name + " was not found.");
+}
+const ov::element::Type& ov::Function::output_element_type(size_t i) const {
+    return m_results.at(i)->get_output_element_type(0);
+}
+const ov::element::Type& ov::Function::output_element_type(const std::string& tensor_name) const {
+    for (const auto& res : m_results) {
+        for (const auto& name : res->get_input_tensor(0).get_names()) {
+            if (name == tensor_name)
+                return res->get_output_element_type(0);
+        }
+    }
+    throw ov::Exception("Output for tensor name " + tensor_name + " was not found.");
+}
+const ov::Shape& ov::Function::output_shape(size_t i) const {
+    return m_results.at(i)->get_output_partial_shape(0);
+}
+const ov::Shape& ov::Function::output_shape(const std::string& tensor_name) const {
+    for (const auto& res : m_results) {
+        for (const auto& name : res->get_input_tensor(0).get_names()) {
+            if (name == tensor_name)
+                return res->get_output_partial_shape(0);
+        }
+    }
+    throw ov::Exception("Output for tensor name " + tensor_name + " was not found.");
+}
+/// Input functions
+ov::Output<ov::Node> ov::Function::input(size_t i) const {
+    return m_parameters.at(i);
+}
+ov::Output<ov::Node> ov::Function::input(const std::string& tensor_name) const {
+    for (const auto& param : m_parameters) {
+        for (const auto& name : param->get_output_tensor(0).get_names()) {
+            if (name == tensor_name)
+                return param;
+        }
+    }
+    throw ov::Exception("Input for tensor name " + tensor_name + " was not found.");
+}
+const ov::element::Type& ov::Function::input_element_type(size_t i) const {
+    return m_parameters.at(i)->get_output_element_type(0);
+}
+const ov::element::Type& ov::Function::input_element_type(const std::string& tensor_name) const {
+    for (const auto& param : m_parameters) {
+        for (const auto& name : param->get_output_tensor(0).get_names()) {
+            if (name == tensor_name)
+                return param->get_output_element_type(0);
+        }
+    }
+    throw ov::Exception("Input for tensor name " + tensor_name + " was not found.");
+}
+const ov::Shape& ov::Function::input_shape(size_t i) const {
+    return m_parameters.at(i)->get_output_partial_shape(0);
+}
+const ov::Shape& ov::Function::input_shape(const std::string& tensor_name) const {
+    for (const auto& param : m_parameters) {
+        for (const auto& name : param->get_output_tensor(0).get_names()) {
+            if (name == tensor_name)
+                return param->get_output_partial_shape(0);
```

Comment:
Similarily for parameters, I think the lookup could be moved to a separate helper.